gdal/ogr/ogrsf_frmts/cad/ogrcadlayer.cpp

```diff
@@ -0,0 +1,639 @@
+    poFeatureDefn->AddFieldDefn( &oLinetypeField );
+
+    OGRFieldDefn  oColorField( "color", OFTIntegerList );
+    poFeatureDefn->AddFieldDefn( &oColorField );
+
+    OGRFieldDefn  oExtendedField( "extentity_data", OFTString );
+    poFeatureDefn->AddFieldDefn( &oExtendedField );
+
+    OGRFieldDefn  oTextField( "text", OFTString );
+    poFeatureDefn->AddFieldDefn( &oTextField );
+
+    auto oAttrTags = poCADLayer.getAttributesTags();
+    for( auto citer = oAttrTags.cbegin(); citer != oAttrTags.cend(); ++citer )
+    {
+        OGRFieldDefn oAttrField( (*citer).c_str(), OFTString );
+        poFeatureDefn->AddFieldDefn( &oAttrField );
+        asFeaturesAttributes.push_back( *citer );
+    }
+
+    // Applying spatial ref info
+    poSpatialRef = poSR;
+    poFeatureDefn->GetGeomFieldDefn(0)->SetSpatialRef( poSR );
+
+    SetDescription( poFeatureDefn->GetName() );
+    poFeatureDefn->Reference();
+}
+
+GIntBig OGRCADLayer::GetFeatureCount( int bForce )
+{
+    if( m_poFilterGeom != NULL || m_poAttrQuery != NULL )
+        return OGRLayer::GetFeatureCount( bForce );
+
+    return poCADLayer.getGeometryCount();
+}
+
+OGRCADLayer::~OGRCADLayer()
+{
+	poFeatureDefn->Release();
+}
+
+void OGRCADLayer::ResetReading()
+{
+	nNextFID = 0;
+}
+
+OGRFeature *OGRCADLayer::GetNextFeature()
+{
+    OGRFeature *poFeature = GetFeature( nNextFID );
+    ++nNextFID;
+
+    if( poFeature == NULL )
+        return( NULL );
+
+    if( ( m_poFilterGeom == NULL ||  FilterGeometry( poFeature->GetGeometryRef() ) )
+        && ( m_poAttrQuery == NULL || m_poAttrQuery->Evaluate( poFeature ) ) )
+    {
+        return poFeature;
+    }
+}
+
+OGRFeature *OGRCADLayer::GetFeature( GIntBig nFID )
+{
+    if( poCADLayer.getGeometryCount() <= nFID
+        || nFID < 0 )
+    {
+        return NULL;
+    }
+
+    OGRFeature  *poFeature = NULL;
+    CADGeometry *poCADGeometry = poCADLayer.getGeometry( nFID );
+
+    if( NULL == poCADGeometry || GetLastErrorCode() != CADErrorCodes::SUCCESS )
+    {
+        CPLError( CE_Failure, CPLE_NotSupported,
+                 "Failed to get geometry with ID = %lld from layer \"%s\". Libopencad errorcode: %d",
+                 nFID, poCADLayer.getName().c_str(), GetLastErrorCode() );
+        return NULL;
+    }
+
+    poFeature = new OGRFeature( poFeatureDefn );
+    poFeature->SetFID( nFID );
+    poFeature->SetField( "thickness", poCADGeometry->getThickness() );
+
+    if( poCADGeometry->getEED().size() != 0 )
+    {
+        std::string sEEDAsOneString = "";
+        for ( std::vector<std::string>::const_iterator
+              iter = poCADGeometry->getEED().cbegin();
+              iter != poCADGeometry->getEED().cend(); ++iter )
+        {
+            sEEDAsOneString += *iter;
+            sEEDAsOneString += ' ';
+        }
+
+        poFeature->SetField( "extentity_data", sEEDAsOneString.c_str() );
+    }
+
+    RGBColor stRGB = poCADGeometry->getColor();
+    GIntBig adRGB[3] { stRGB.R, stRGB.G, stRGB.B };
```

Comment:
We discussed with Alexander and decide to store color as hex - #RRGGBB[AA] in string of length 10. QGIS renderer read the values from this field also  it is rather simple to parse.